### ISSUE TITLE
Support: invalid search keyword query should have a hard timeout 

### ISSUE THREAD

**Issue by worldomonation** [Automattic/wp-calypso#55478]
### Steps to Reproduce

1. start at /home.
2. in the support card, search for an invalid term like `;;;ppp;;;`.
3. wait until the request is resolved.

### What I expected to happen

The text `Sorry, there were no matches. Here are some of the most searched for help pages for this section` should appear within a reasonable amount of time.

### What actually happened

The placeholder animation continues for well over 30 seconds.

Occurs frequently on E2E CI tests and less frequently under user interaction.

### Context

Playwright E2E testing.

### Operating System

Linux

### Browser

Chrome/Chromium

### Is this specific to applied theme? If so, what is the theme name?

No

### Simple/Atomic

Simple, Atomic

### Number of Users Impacted

Some users (<50%)

### Available Workarounds

Refresh and try again.

### Reproducibility

Intermittent

### Other information

Suggestion: the amount of time for a query should be capped at some value (eg. 30000ms) instead of allowed to run forever.

**Comment by worldomonation** [comment]
Continues to be reproducible and affecting end-to-end tests on `trunk` and elsewhere.

- https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Playwright_mobile/6705169?buildTab=tests&expandedTest=build%3A%28id%3A6705169%29%2Cid%3A5485#%2Fscreenshots
- https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Playwright_mobile/6702607
- https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Playwright_mobile/6699568
- https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Playwright_desktop/6706277
- https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Playwright_desktop/6705180?buildTab=tests&status=muted&expandedTest=build%3A%28id%3A6705180%29%2Cid%3A5311

If this continues, the `Invalid keyword` test case should be skipped temporarily.

**Comment by dpasque** [comment]
It looks like a completely empty search of just whitespace characters has also started to take a decent amount of time as well.

**Comment by cuemarie** [comment]
**Replicated?** ✅ - I was able to trip up the Support Card search with lots of spaces, but not with `;;;ppp;;;`. Still, this shows where a hard timeout would improve the experience overall

**Browsers Tested:** Firefox, Chrome (on Mac OS)
**Environments Tested:** AT, Simple

**Notes:**

**;;;ppp;;;**
Searched for `;;;ppp;;;` in FireFox and Chrome on Simple and AT sites, and the longest I had to wait was 10 seconds to see `Sorry, there were no matches. Here are some of the most searched for help pages for this section:`
https://user-images.githubusercontent.com/27249804/164092189-d6aa30bb-5ea3-4da2-ac2f-515e7e8e27c8.mp4

**URL / other characters**
I also tried searching a droplr URL (by accident) and other random characters, and the wait time was roughly the same, under 10 seconds.
https://user-images.githubusercontent.com/27249804/164092177-58db0c76-cc09-4957-9b42-1b5b411dc0e0.mp4

**Empty Spaces**
Searching for 15-20 space characters did push the browser to take longer, though this seemed in part because the search card starting looking for results right away. Couldn't get super consistent results here - mostly the search results appeared in the same general time.

**Longest Wait:**

https://user-images.githubusercontent.com/27249804/164092017-d127a79c-9a22-47ca-911e-dabeffbf8daf.mp4

**Comment by cuemarie** [comment]
@worldomonation Since this original report is a bit old, I wonder if the original invalid-characters issue is solved now. Can you test again on your end to see if you're still running into the issue?

@dpasque thanks for moving this to the triaging board. I had some success replicating, but it wasn't very consistent. Are you seeing more consistent wait times when running 'a completely empty search of just whitespace characters'?

**Comment by dpasque** [comment]
>I had some success replicating, but it wasn't very consistent. Are you seeing more consistent wait times when running 'a completely empty search of just whitespace characters'?

@cuemarie  I have a somewhat similar experience. It seems vaguely related to the amount of whitespace characters but also varies a lot in its execution time. I just got one that took about 20-30 seconds on the home page. I effectively did this just by holding down the space bar on the search field.

![whitespace_search](https://user-images.githubusercontent.com/26772342/164240362-a7209908-7f0b-4900-96b9-9ce6ac78a761.gif)

**Comment by dpasque** [comment]
The practical use case of this isn't particularly relevant. But it does make me wonder if there's some part of our underlying search algorithm that's doing something accidentally expensive. 🤔

**Comment by worldomonation** [comment]
Thanks for testing @dpasque! 

I also gave it a whirl using Firefox Nightly 101.0a1 (2022-04-19) (64-bit) and observed largely what Dan did - that is, the message appears much quicker than it used to when the report was filed.

I typed a 207-character sequence of space characters, split into three roughly equal segments (hit spacebar for some time, wait a second to let the animation play and repeat twice more). Even with the intentional attempt I was not able to trigger the lookup to last more than 7-10 seconds.

With the information above, I would say that this is a much lower priority than it used to be.
